src/iot/fieldGroupings.js

```diff
@@ -0,0 +1,327 @@
+      if (!fieldGrouping[field]) {
+        return Promise.reject(
+          new Error(`A ${field} is required to create a new field grouping.`)
+        );
+      }
+    }
+
+    const formattedGrouping = toSnakeCase(fieldGrouping);
+
+    return this._request
+      .post(
+        `${this._baseUrl}/facilities/${facilityId}/groupings`,
+        formattedGrouping
+      )
+      .then((fieldGrouping) => toCamelCase(fieldGrouping));
+  }
+
+  /**
+   * Deletes a field grouping
+   *
+   * API Endpoint: '/groupings/:groupingId'
+   * Method: DELETE
+   *
+   * @param {String} groupingId The UUID of a field grouping
+   *
+   * @returns {Promise}
+   * @fulfill {undefined}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.iot.fieldGroupings
+   *   .delete('b3dbaae3-25dd-475b-80dc-66296630a8d0');
+   */
+  delete(groupingId) {
+    if (!groupingId) {
+      return Promise.reject(
+        new Error('A groupingId is required for deleting a field grouping.')
+      );
+    }
+
+    return this._request.delete(`${this._baseUrl}/groupings/${groupingId}`);
+  }
+
+  /**
+   * Gets information about a field grouping
+   *
+   * API Endpoint: '/groupings/:groupingId'
+   * Method: GET
+   *
+   * @param {String} groupingId The UUID of a field grouping
+   *
+   * @returns {Promise}
+   * @fulfill {FieldGrouping} Information about the field grouping
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.iot.fieldGroupings
+   *   .get('b3dbaae3-25dd-475b-80dc-66296630a8d0')
+   *   .then((fieldGrouping) => console.log(fieldGrouping))
+   *   .catch((err) => console.log(err));
+   */
+  get(groupingId) {
+    if (!groupingId) {
+      return Promise.reject(
+        new Error(
+          'A groupingId is required for getting information about a field grouping.'
+        )
+      );
+    }
+
+    return this._request
+      .get(`${this._baseUrl}/groupings/${groupingId}`)
+      .then((fieldGrouping) => toCamelCase(fieldGrouping));
+  }
+
+  /**
+   * Get a listing of all field groupings for a facility available to the user. Includes public groupings across
+   * any organization the user has access to and the user's private groupings.
+   *
+   * API Endpoint: '/facilities/:facilityId/groupings'
+   * Method: GET
+   *
+   * @param {number} facilityId The ID of a facility with groupings
+   *
+   * @returns {Promise}
+   * @fulfill {FieldGrouping[]} Information about the field grouping
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.iot.fieldGroupings
+   *   .getAll(135)
+   *   .then((fieldGroupings) => console.log(fieldGroupings))
+   *   .catch((err) => console.log(err));
+   */
+  getAll(facilityId) {
+    if (!facilityId) {
+      return Promise.reject(
+        new Error('A facilityId is required for getting all field groupings.')
+      );
+    }
+
+    return this._request
+      .get(`${this._baseUrl}/facilities/${facilityId}/groupings`)
+      .then((fieldGrouping) => toCamelCase(fieldGrouping));
```

Comment:
This endpoint is also paginated. Same comments apply here as in the fieldCategories file.